# ç»¼åˆè‡ªåŠ¨åŒ–æ£€æŸ¥æŠ¥å‘Š

**ç”Ÿæˆæ—¶é—´**: 2025-10-11T03:49:22.865Z

## ğŸ“Š æ£€æŸ¥æ‘˜è¦

| çŠ¶æ€    | æ•°é‡ | ç™¾åˆ†æ¯” |
| ------- | ---- | ------ |
| âœ… é€šè¿‡ | 3    | 43%    |
| âš ï¸ è­¦å‘Š | 1    | 14%    |
| âŒ å¤±è´¥ | 3    | 43%    |
| ğŸ“Š æ€»è®¡ | 7    | 100%   |

**æˆåŠŸç‡**: 43%

## ğŸ“‹ è¯¦ç»†ç»“æœ

| æ£€æŸ¥é¡¹ç›®               | çŠ¶æ€ | è€—æ—¶    | å‘½ä»¤                      |
| ---------------------- | ---- | ------- | ------------------------- |
| âœ… ç¯å¢ƒæ£€æŸ¥            | é€šè¿‡ | 9338ms  | `pnpm run check:env`      |
| âœ… å®‰å…¨å®¡è®¡            | é€šè¿‡ | 7345ms  | `pnpm run check:security` |
| âœ… ä»£ç è´¨é‡æ£€æŸ¥        | é€šè¿‡ | 95263ms | `pnpm run check:quality`  |
| âš ï¸ ä»£ç è§„èŒƒæ£€æŸ¥        | è­¦å‘Š | 3663ms  | `pnpm run lint`           |
| âŒ ä»£ç æ ¼å¼åŒ–æ£€æŸ¥      | å¤±è´¥ | 5203ms  | `pnpm run format:check`   |
| âŒ TypeScript ç±»å‹æ£€æŸ¥ | å¤±è´¥ | 3273ms  | `npx tsc --noEmit`        |
| âŒ æµ‹è¯•è¿è¡Œ            | å¤±è´¥ | 14423ms | `pnpm run test`           |

## âŒ å¤±è´¥æ£€æŸ¥è¯¦æƒ…

### ä»£ç æ ¼å¼åŒ–æ£€æŸ¥

**å‘½ä»¤**: `pnpm run format:check`

**é”™è¯¯ä¿¡æ¯**:

```
[warn] apps/api/api-extractor.json
[warn] apps/api/docs/api/api.json
[warn] apps/api/docs/api/fastify-api.api.md
[warn] apps/api/jest.setup.js
[warn] apps/api/package.json
[warn] apps/api/README.md
[warn] apps/api/src/app.ts
[warn] apps/api/src/routes/user.route.ts
[warn] apps/api/tests/integration/user.integration.test.ts
[warn] apps/api/tests/unit/user.service.test.ts
[warn] apps/api/tsconfig.json
[warn] apps/api/tsdoc-metadata.json
[warn] apps/web/package.json
[warn] apps/web/src/App.tsx
[warn] apps/web/src/test/App.test.tsx
[warn] apps/web/src/test/setup.ts
[warn] apps/web/vitest.config.ts
[warn] commitlint.config.js
[warn] docs/api/openapi.json
[warn] docs/api/README.md
[warn] docs/api/swagger-setup.md
[warn] docs/database/README.md
[warn] docs/development/dev-tools.md
[warn] docs/generated/analysis/dependency-analysis.md
[warn] docs/generated/analysis/project-structure.md
[warn] docs/generated/changelog/changelog.md
[warn] docs/generated/cicd/cicd-pipeline.md
[warn] docs/generated/config/docker-config.md
[warn] docs/generated/config/environment-config.md
[warn] docs/generated/database/database-design.md
[warn] docs/generated/README.md
[warn] docs/generated/reports/checks/code-quality.json
[warn] docs/generated/reports/checks/code-quality.md
[warn] docs/generated/reports/checks/environment-check.json
[warn] docs/generated/reports/checks/security-audit.json
[warn] docs/generated/reports/checks/security-audit.md
[warn] docs/generated/reports/duplicate-dir-fix-report.json
[warn] docs/generated/reports/file-organization-report.json
[warn] docs/generated/reports/naming-standardization-report.json
[warn] docs/generated/reports/path-update-report.json
[warn] docs/generated/reports/project-health.md
[warn] docs/generated/testing/test-documentation.md
[warn] eslint.config.js
[warn] infrastructure/docker/docker-compose.yml
[warn] infrastructure/README.md
[warn] package.json
[warn] performance-config.json
[warn] README.md
[warn] services/grpc/README.md
[warn] tests/README.md
[warn] tools/scripts/automation/check-environment.js
[warn] tools/scripts/automation/code-quality-check.js
[warn] tools/scripts/automation/run-all-checks.js
[warn] tools/scripts/automation/security-audit.js
[warn] tools/scripts/development/generate-all-docs.js
[warn] tools/scripts/development/generate-docs.js
[warn] tools/scripts/development/restore-after-clean.js
[warn] tools/scripts/maintenance/clean.js
[warn] tools/scripts/maintenance/maintenance.js
[warn] tools/scripts/monitoring/monitoring-log.js
[warn] tools/scripts/monitoring/performance-test.js
[warn] Code style issues found in 61 files. Run Prettier with --write to fix.

```

**å»ºè®®**: è¿è¡Œ pnpm run format æ ¼å¼åŒ–ä»£ç 

### TypeScript ç±»å‹æ£€æŸ¥

**å‘½ä»¤**: `npx tsc --noEmit`

**é”™è¯¯ä¿¡æ¯**:

```
npm warn Unknown env config "verify-deps-before-run". This will stop working in the next major version of npm.
npm warn Unknown env config "_jsr-registry". This will stop working in the next major version of npm.

```

**å»ºè®®**: ä¿®å¤ TypeScript ç±»å‹é”™è¯¯

### æµ‹è¯•è¿è¡Œ

**å‘½ä»¤**: `pnpm run test`

**é”™è¯¯ä¿¡æ¯**:

```
npm warn Unknown env config "verify-deps-before-run". This will stop working in the next major version of npm.
npm warn Unknown env config "_jsr-registry". This will stop working in the next major version of npm.
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
See more at https://kulshekhar.github.io/ts-jest/docs/getting-started/presets#advanced
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
See more at https://kulshekhar.github.io/ts-jest/docs/getting-started/presets#advanced
FAIL tests/integration/user.integration.test.ts
  â— User API Integration Tests â€º GET /users â€º should return users list

    TypeError: (0 , app_1.build) is not a function

       6 |
       7 |     beforeAll(async () => {
    >  8 |         app = build();
         |                    ^
       9 |         await app.ready();
      10 |     });
      11 |

      at Object.<anonymous> (tests/integration/user.integration.test.ts:8:20)

  â— User API Integration Tests â€º POST /users â€º should create a new user

    TypeError: (0 , app_1.build) is not a function

       6 |
       7 |     beforeAll(async () => {
    >  8 |         app = build();
         |                    ^
       9 |         await app.ready();
      10 |     });
      11 |

      at Object.<anonymous> (tests/integration/user.integration.test.ts:8:20)

  â— User API Integration Tests â€º POST /users â€º should return 400 for invalid data

    TypeError: (0 , app_1.build) is not a function

       6 |
       7 |     beforeAll(async () => {
    >  8 |         app = build();
         |                    ^
       9 |         await app.ready();
      10 |     });
      11 |

      at Object.<anonymous> (tests/integration/user.integration.test.ts:8:20)


  â— Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'close')

      11 |
      12 |     afterAll(async () => {
    > 13 |         await app.close();
         |                   ^
      14 |     });
      15 |
      16 |     describe('GET /users', () => {

      at Object.<anonymous> (tests/integration/user.integration.test.ts:13:19)

FAIL tests/unit/user.service.test.ts (9.377 s)
  â— UserService â€º create â€º should create a user with valid data

    PrismaClientInitializationError:
    Invalid `prisma.user.create()` invocation in
    D:\WorkSpace\ProjectDevelopment\fastify-react-app-enhance\apps\api\src\services\user.service.ts:5:17

      2 export const userService = {
      3   getAll: async () => prisma.user.findMany(),
      4   create: async (data: { name: string; email: string }) =>
    â†’ 5     prisma.user.create(
    Can't reach database server at `localhost:5432`

    Please make sure your database server is running at `localhost:5432`.

       9 |             };
      10 |
    > 11 |             const result = await userService.create(userData);
         |                            ^
      12 |
      13 |             expect(result).toBeDefined();
      14 |             expect(result.name).toBe(userData.name);

      at $n.handleRequestError (../../node_modules/.pnpm/@prisma+client@5.22.0_prisma@5.22.0/node_modules/@prisma/client/runtime/library.js:121:7615)
      at $n.handleAndLogRequestError (../../node_modules/.pnpm/@prisma+client@5.22.0_prisma@5.22.0/node_modules/@prisma/client/runtime/library.js:121:6623)
      at $n.request (../../node_modules/.pnpm/@prisma+client@5.22.0_prisma@5.22.0/node_modules/@prisma/client/runtime/library.js:121:6307)
      at l (../../node_modules/.pnpm/@prisma+client@5.22.0_prisma@5.22.0/node_modules/@prisma/client/runtime/library.js:130:9633)
      at Object.<anonymous> (tests/unit/user.service.test.ts:11:28)

  â— UserService â€º findAll â€º should return an array of users

    TypeError: user_service_1.userService.findAll is not a function

      28 |     describe('findAll', () => {
      29 |         it('should return an array of users', async () => {
    > 30 |             const result = await userService.findAll();
         |                                              ^
      31 |
      32 |             expect(Array.isArray(result)).toBe(true);
      33 |         });

      at Object.<anonymous> (tests/unit/user.service.test.ts:30:46)

Test Suites: 2 failed, 2 total
Tests:       5 failed, 1 passed, 6 total
Snapshots:   0 total
Time:        11.06 s
Ran all test suites.
npm error Lifecycle script `test` failed with error:
npm error code 1
npm error path D:\WorkSpace\ProjectDevelopment\fastify-react-app-enhance\apps\api
npm error workspace fastify-api@0.1.0
npm error location D:\WorkSpace\ProjectDevelopment\fastify-react-app-enhance\apps\api
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c jest

```

**å»ºè®®**: ä¿®å¤å¤±è´¥çš„æµ‹è¯•ç”¨ä¾‹

## âš ï¸ è­¦å‘Šæ£€æŸ¥è¯¦æƒ…

### ä»£ç è§„èŒƒæ£€æŸ¥

**å‘½ä»¤**: `pnpm run lint`

**è­¦å‘Šä¿¡æ¯**:

```
Command failed: pnpm run lint
```

**å»ºè®®**: è¿è¡Œ pnpm run lint:fix è‡ªåŠ¨ä¿®å¤

## ğŸ’¡ ä¿®å¤å»ºè®®

```bash
# è‡ªåŠ¨ä¿®å¤ä»£ç è§„èŒƒé—®é¢˜
pnpm run lint:fix

# æ ¼å¼åŒ–ä»£ç 
pnpm run format

# è¿è¡Œæµ‹è¯•
pnpm run test

# å®‰å…¨å®¡è®¡
pnpm run check:security

# ä»£ç è´¨é‡æ£€æŸ¥
pnpm run check:quality
```

---

_æ­¤æŠ¥å‘Šç”±ç»¼åˆè‡ªåŠ¨åŒ–æ£€æŸ¥è„šæœ¬ç”Ÿæˆ_
